typetest/fake-bundle/src/dashboard/index.ts

```diff
@@ -12,6 +12,26 @@ nodecg.Logger.globalReconfigure({
 logger.error('shaking my smh');
 nodecg.log.trace('some verbose logs here');
 
+const rep1 = nodecg.Replicant('rep1');
+const rep2 = nodecg.Replicant('rep2');
+const rep3 = nodecg.Replicant('rep3');
+
+NodeCG.waitForReplicant(rep1).then(() => {
+	console.log('[static] rep1 is fully declared and ready to use!');
+});
+
+NodeCG.waitForReplicants(rep2, [rep3]).then(() => {
+	console.log('[static] rep2 and rep3 are fully declared and ready to use!');
+});
+
+nodecg.waitForReplicant(rep1).then(() => {
+	console.log('[instance] rep1 is fully declared and ready to use!');
+});
+
+nodecg.waitForReplicants(rep2, [rep3]).then(() => {
```

Comment:
I would prefer leaving users the choice to choose the syntax they want. I personally prefer working with arrays, simpler than spreading.
Also, I don't see why the TypeScript code would be more complex only because of a array flattening. I translated into TS myself and nothing complicated AFAIK: https://bit.ly/2ryLRxW

Reply:
It’s simpler, more readable, more stable and more consistent to allow only spread parameters or array of replicants.
Mixing those up hurts both performance (doing concat thing even if there’s no array), and readability (`waitForReplicants(rep, reps, reps, rep, reps,...)`).
If you need to handle arrays and non-arrays all you have to do is put `...` on arrays.

Reply:
I’d like to hear @Lange ‘a opinion on this.

Reply:
I admit that this does seem like a lot of work for something which is already handled by the spread operator. Are other users asking for this? If so, are they aware of the spread operator?